app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -112,25 +112,50 @@ if [ x$ID = xrhel ] || [ x$ID = xcentos ]; then
 <%= "          --data 'remote_execution_interface=#{@remote_execution_interface}' \\\n" if @remote_execution_interface.present? -%>
 <%= "          --data 'packages=#{@packages}' \\\n" if @packages.present? -%>
 
-}
+    }
 
     <% if @force -%>
-    yum remove -y katello-ca-consumer*
+    yum remove -y katello-ca-consumer* subscription-manager
```

Comment:
I think with this change you can drop this whole force block.

Reply:
Additionally, if this template is what configured subscription-manager, I lean towards removing any installed consumer RPMs off the system so as not to give a false impression that they are affecting the client.

Reply:
This is only for case when user run registration with `force=true`, which he/she knows that deletes these files. It was also requested by support in BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1921055

Reply:
I am leaning towards we should be removing `katello-ca-consumer*` regardless of whether a parameter is provided. This is about registration, so we should assume that said system is *not* registered or configured correctly at the time the user wants to register the system and cleanup any potential issues on the system.

The idea of a `force` parameter is a holdover concept from subscription-manager where if you were on the client and wanted to either change the smart-proxy it was registered to or change the Content View / Lifecycle environment you had to use the `--force` parameter to basically "recreate" the system. I think we should evolve to look at registration as bringing a machine previously unknown to Foreman to Foreman's inventory and thus cleanup any potential old stuff we know might exist and cause problems and lay down proper configuration based on the users parameters. For changes that happen after registration, that's what other stories and mechanisms should be used for (e.g. changing content source, using REX).